polytracker/src/passes/polytracker_pass.cpp

```diff
@@ -134,12 +134,29 @@ void PolytrackerPass::visitBranchInst(llvm::BranchInst &BI) {
   }
 }
 
-void PolytrackerPass::visitSwitchInst(llvm::SwitchInst &SI) {
-  // TODO (hbrodin): Is a guard needed? E.g. op_check as in visitBranchInst?
-  llvm::IRBuilder<> IRB(&SI);
+void PolytrackerPass::visitGetElementPtrInst(llvm::GetElementPtrInst &GEP) {
+
+  llvm::IRBuilder<> IRB(&GEP);
   llvm::LLVMContext &context = mod->getContext();
-  llvm::Value *int_val = IRB.CreateSExtOrTrunc(SI.getCondition(), shadow_type);
-  CallInst *Call = IRB.CreateCall(conditional_branch_log, {int_val});
+  for (auto &op : GEP.indices()) {
+    // Ignore constant indices
+    if (!llvm::isa<llvm::ConstantInt>(op)) {
+      if (!op_check(op)) {
+        llvm::Value *int_val = IRB.CreateSExtOrTrunc(op, shadow_type);
+        CallInst *Call = IRB.CreateCall(conditional_branch_log, {int_val});
+      }
+    }
+  }
+}
+
+void PolytrackerPass::visitSwitchInst(llvm::SwitchInst &SI) {
```

Comment:
Correct me if I'm wrong, but haven't we found out that we don't need this code because dfsan already instruments switches?

Reply:
There is instrumentation for select, but none for switch. We need the switch to detect control flow influence (same as for conditional branches). DFSAN do not instrument branches either.